coredocument/coredocument.proto

```diff
@@ -94,3 +102,49 @@ message Signature {
   bytes signature = 3;
   google.protobuf.Timestamp timestamp = 4;
 }
+
+// Action defines the set of actions a collaborator can/have per document.
+enum Action {
+  ACTION_INVALID = 0;
+  // read represents just reading the doc/fields
+  ACTION_READ = 1;
+  // read_sign represents reading as well the sign the documents. We will pick this one when requesting the signatures.
+  ACTION_READ_SIGN = 2;
+}
+
+// Roles holds a list of collaborators, NFTs, and/or access tokens.
+message Role {
+  // label to identify the role
+  uint32 label = 1;
+  // collaborators holds the list of document collaborators
+  repeated bytes collaborators = 3;
+  // nfts is a list of registry address/tokenID pairs.
+  // For easier verification in merkle proofs, the values are simply concatenated with the first 22 bytes being the NFT registry and the remaining 32 bytes the tokenID.
+  repeated bytes nfts = 4;
+  bytes access_token_identifier = 5;
+}
+
+message SaltedRole {
+  bytes label = 1;
+  repeated bytes collaborators = 2;
+  bytes collaborators_length = 3;
+  repeated bytes nfts = 4;
+  bytes nfts_length = 5;
+  bytes access_token_identifier = 6;
+}
+
+message ReadRule {
+  // label to identify the ReadRule
```

Comment:
I'd specify that labels are purely for the convenience of the user and are not used by the protocol for anything.

Reply:
same for label in role